engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java

```diff
@@ -147,30 +137,16 @@ protected Changes applyDiff(S from, S to, RuntimeException e) {
     public synchronized final Changes evaluateScript(final String script, final @Nullable String scriptName) {
         RuntimeException evaluateErr = null;
         final Changes diff;
-        try (S fromSnapshot = takeSnapshot()) {
-
-            // store pointers to exist query scope static variables
-            final QueryLibrary prevQueryLibrary = QueryLibrary.getLibrary();
-            final CompilerTools.Context prevCompilerContext = CompilerTools.getContext();
-            final QueryScope prevQueryScope = QueryScope.getScope();
-
-            // retain any objects which are created in the executed code, we'll release them when the script session
-            // closes
-            try (final SafeCloseable ignored = LivenessScopeStack.open(this, false)) {
-                // point query scope static state to our session's state
-                QueryScope.setScope(queryScope);
-                CompilerTools.setContext(compilerContext);
-                QueryLibrary.setLibrary(queryLibrary);
+        // retain any objects which are created in the executed code, we'll release them when the script session
+        // closes
+        try (S fromSnapshot = takeSnapshot();
+                final SafeCloseable ignored = LivenessScopeStack.open(this, false)) {
 
+            try {
                 // actually evaluate the script
-                evaluate(script, scriptName);
+                executionContext.apply(() -> evaluate(script, scriptName));
```

Comment:
I think this needs to be applied for the snapshots, since they are capturing QueryScope.

Reply:
I think Snapshots are already OK. They are capturing via the `GroovyShell` and `PythonScope` directly without asking the `QueryScope` thread local value.